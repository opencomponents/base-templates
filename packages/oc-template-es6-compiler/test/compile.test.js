jest.setTimeout(100000);

const path = require('path');
const compile = require('../lib/compile.js');
const fs = require('fs-extra');
const nodeDir = require('node-dir');

const componentPackage = require('../../../mocks/es6-component/package.json');
const componentPath = path.join(__dirname, '../../../mocks/es6-component/');
const getInfo = require('../index.js').getInfo;
const publishPath = path.join(componentPath, '__package');

test('Should correctly compile the oc component', done => {
  const options = {
    componentPackage,
    ocPackage: {
      version: '1.0.0'
    },
    componentPath,
    publishPath,
    production: true
  };
  fs.ensureDirSync(publishPath);
  compile(options, (err, res) => {
    expect(err).toBeNull();
    const version = res.oc.files.template.version;
    res.oc.files.template.version = '';
    res.oc.date = '';
    res.oc.files.dataProvider.hashKey = 'dummyData';
    res.oc.files.template.hashKey = 'dummyData';
    expect(res).toMatchSnapshot();
    expect(version).toBe(getInfo().version);
    nodeDir.paths(publishPath, (err, res) => {
      const files = res.files
        .map(filePath => path.relative(__dirname, filePath))
        .sort();
      expect(files).toMatchSnapshot();
      fs.emptyDirSync(publishPath);
      done();
    });
  });
});
