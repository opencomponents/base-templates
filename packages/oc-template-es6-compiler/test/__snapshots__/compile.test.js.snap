// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly compile the oc component 1`] = `
Object {
  "dependencies": Object {
    "lodash": "",
  },
  "description": "",
  "devDependencies": Object {
    "oc-template-es6-compiler": "",
  },
  "name": "es6-component",
  "oc": Object {
    "date": "",
    "files": Object {
      "dataProvider": Object {
        "hashKey": "dummyData",
        "src": "server.js",
        "type": "node.js",
      },
      "static": Array [],
      "template": Object {
        "hashKey": "dummyData",
        "src": "template.js",
        "type": "oc-template-es6",
        "version": "",
      },
    },
    "packaged": true,
    "version": "1.0.0",
  },
  "repository": "",
  "version": "1.0.0",
}
`;

exports[`Should correctly compile the oc component 2`] = `
Array [
  "../../../mocks/es6-component/__package/package.json",
  "../../../mocks/es6-component/__package/server.js",
  "../../../mocks/es6-component/__package/styles.css",
  "../../../mocks/es6-component/__package/template.js",
]
`;
